### PR TITLE
Add Husky and Travis

### LINKED ISSUE #104: Setup pre-commit lint hooks with Husky

### PR DESCRIPTION
**Pull request by mark-nakachon**
### Checklist:

* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [x] Have you successfully ran tests with your changes locally?
* [x] Have you added an explanation of what your changes do and why you'd like us to include them?

### Issues Solved

Fixes #104

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+trim_trailing_whitespace = true
+
+# Unix-style newlines with a newline ending every file
+[*]
+indent_style = space
+indent_size = 2
+end_of_line = lf
+charset = utf-8
+insert_final_newline = true
+max_line_length = 100
```
Comment (aashutoshrathi) [inline .editorconfig]: Newline at the end

File: .eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    "globals": {
+        "Atomics": "readonly",
+        "SharedArrayBuffer": "readonly"
+    },
+    "parserOptions": {
+        "ecmaVersion": 2018
+    },
+    "rules": {
+    }
+}
```
Comment (aashutoshrathi) [inline .eslintrc.json]: Same here.

**Comment by aashutoshrathi** [comment]
Also, Travis is missing and there are conflicts, please resolve them

File: .editorconfig [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
```
Comment (aashutoshrathi) [inline .editorconfig]: You added one extra line :confused:

File: .eslintrc.json [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
```
Comment (aashutoshrathi) [inline .eslintrc.json]: Newline should be empty

**Comment by mark-nakachon** [comment]
@aashutoshrathi @rossoskull there is already Travis CI added to the project an I can also see the  travis.yml file what should I add ??

**Comment by aashutoshrathi** [comment]
@mark-nakachon You have to check for lint on every build, if the code is not linted then build should fail.

**Comment by aashutoshrathi** [comment]
You need to change the environment and install node dependencies.

**Comment by aashutoshrathi** [comment]
Also, fix the lint errors raised.

**Comment by mark-nakachon** [comment]
Fixed all the errors and the build was successful now

**Pull request merged**